### PR TITLE
Use babel-mapbox-preset

### PR DESCRIPTION
**Pull request by kepta**
- This moves the `babel-mapbox-preset` to packages. I haven't yet made it a monorepo yet, using relative paths. 
- Also added code to check if `babelrc` exists use that.
- In the coming PR's I will add the ability to pass in browserslist to `babelrc`.

@davidtheclark for review.

### CODE REVIEW

File: lib/create-webpack-config.js [hunk 1]
```diff
@@ -120,4 +106,23 @@ function createWebpackConfig(urc) {
...
+  const loader = {
+    loader: 'babel-loader',
+    options: {
+      presets: [require.resolve('../packages/babel-preset-mapbox')],
+      cacheDirectory: true,
+      babelrc: false,
+      compact: false
+    }
+  };
+  const exists = fse.pathExistsSync(path.join(urc.rootDirectory, '.babelrc'));
```
Comment (davidtheclark) [inline lib/create-webpack-config.js]: `fs.existsSync` is a function. Do you think we can cut the `fs-extra` dependency?

File: packages/babel-preset-mapbox/index.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+module.exports = function preset(context, opts) {
+  opts = opts || [];
```
Comment (davidtheclark) [inline packages/babel-preset-mapbox/index.js]: How can the user pass in a `browserslist` value?

Comment (kepta) [reply packages/babel-preset-mapbox/index.js]: So the idea is to get `browserslist` from the `.babelrc` or else use a default value.

As per our last discussion `babel-preset-mapbox` would provide a direct way for user to provide browserslist: 
```
{
   presets: [["babel-preset-mapbox", [
    "last 1 version",
    "> 1%",
    "maintained node versions",
    "not dead"
  ]]
}
```
or 
```
{
   presets: [["babel-preset-mapbox", [
   ["browserslist", ["last 1 version",
    "> 1%",
    "maintained node versions",
    "not dead"
  ]]]
}
```
(Note: this is not yet implemented)

**Comment by kepta** [comment]
Going to go ahead and merge this, will send another PR to 
- Make the babel preset accept an object as a parameter
- Underreact is able to read babelrc's browserlist option

**Pull request merged**